### ISSUE TITLE
macosx backend does not work with blitting

### ISSUE THREAD

**Issue by WeatherGod** [matplotlib/matplotlib#531]
Looks like the FigureCanvasMac object for the macosx backend have a few missing functions that are critical for animations.  First, it doesn't have copy_from_bbox().  Besides animations, this also impacts some widgets, particularly the lasso widget.  The FigureCanvasMac object also doesn't have restore_region(), so any animations that have blit=True will fail.

**Comment by raw915** [comment]
Here it is April 2015 and animation capability on mac OSX 10.10.2 is still not working. Any hope? Will this be fixed in v1.5.x?

**Comment by mdehoon** [comment]
@raw915  To solve this bug, the animations code will have to be redesigned. I did not write the animations code and I don't have time to fix it myself. If you can volunteer to work on this, or find somebody willing to take on this task, I'd be happy to explain what is wrong with the current code.

**Comment by raw915** [comment]
I used the suggestion from http://stackoverflow.com/users/2388218/matt at http://stackoverflow.com/questions/9401658/matplotlib-animating-a-scatter-plot to get the first example working on OSX 10.10.2. He inserted import matplotlib and matplotlib.use('TkAgg') at the top of the file. This may not work if Tk/Tcl is not installed. I have it through Anaconda https://store.continuum.io/cshop/anaconda/

**Comment by zhanglongqi** [comment]
same issue +1

**Comment by michaelpacer** [comment]
Just wanted to chime in that I ran into this as well. It's kinda intimidating that the default simple example (http://matplotlib.org/1.4.1/examples/animation/simple_anim.html) doesn't work out of the box.

If I have time I'll try to look at the underlying animations code, but I'm no expert so I don't know if I'll actually be able to help.

**Comment by nils-werner** [comment]
Just a fyi, I've refined my function definition a bit more:

```bash
export GLOBALPYTHON=$(which python)

function python {
    if [[ ! -z "$VIRTUAL_ENV" ]]; then
        #test if we are running python 2 or 3
        pyversion=`command python -V 2>&1 | cut -d ' '  -f2 | cut -d. -f1,2`
        PYTHONHOME=$VIRTUAL_ENV $GLOBALPYTHON$pyversion "$@"
    else
        $GLOBALPYTHON "$@"
    fi
}
```

put that in your `.bashrc` and running `python` afterwards should "just work".

**Comment by efiring** [comment]
I changed this to "unassigned" because the solution to problems such as this will likely involve major changes such as an agg-based version of macosx.  The immediate workaround is to use one of the existing backends other than macosx.

**Comment by mdboom** [comment]
Fixed by #6178

**Comment by ReaddyEddy** [comment]
As noted at https://mail.python.org/pipermail/pythonmac-sig/2012-September/023664.html use: 

import matplotlib
matplotlib.use('TkAgg')

just *before*

import numpy as np
import matplotlib.pyplot as plt
import matplotlib.animation as animation

This has worked for me with Tkinter installed using the ActiveState Tkinter installation on OSX 10.11.6, Python 2.71 
The basic animation example is still a little noisy until blt=False in the line_ani code here:
line_ani = animation.FuncAnimation(fig1, update_line, 25, fargs=(data, l),
    interval=50, blit=False)
